### ISSUE TITLE
Slideshow format: Uncaught Error: Unknown dependency: mediawiki.legacy.ajax

### ISSUE THREAD

**Issue by jaideraf** [SemanticMediaWiki/SemanticResultFormats#137]
```
MediaWiki: 1.26.2 (f465524)
PHP: 5.6.14 (cgi-fcgi)
Semantic MediaWiki: 2.4-alpha (560783f)
Semantic Result Formats: 2.3 (ca718a7)
```

When I use Slideshow Format the following issue appears, breaking other javaScript things on the page:

```
Uncaught Error: Unknown dependency: mediawiki.legacy.ajax
load.php?debug=false&lang=pt-br&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=%2BXvto…:160
```

**Comment by kghbln** [comment]
> When I use Slideshow Format the following issue appears, breaking other javaScript things on the page:

Now after upgrading from MW 1.23 to MW 1.27 I can confirm. However "slideshow" is not working either.

```
Error: Unknown dependency: mediawiki.legacy.ajax  load.php:160:232
	sortDependencies https://www.semantic-mediawiki.org/w/load.php:160:232
	sortDependencies https://www.semantic-mediawiki.org/w/load.php:161:41
	resolve/< https://www.semantic-mediawiki.org/w/load.php:161:190
	.each https://www.semantic-mediawiki.org/w/load.php:5:226
	resolve https://www.semantic-mediawiki.org/w/load.php:161:154
	mw.loader.load https://www.semantic-mediawiki.org/w/load.php:172:80
	<anonym> https://www.semantic-mediawiki.org/wiki/Help:Slideshow_format:531:101
	startUp https://www.semantic-mediawiki.org/w/load.php:35:206
	script.onreadystatechange https://www.semantic-mediawiki.org/w/load.php:35:621
```

**Comment by kghbln** [comment]
@s7eph4n As the original author of the "slideshow" format: I am not sure if you are aware of this issue. The [ajax was removed in MW 1.26](https://phabricator.wikimedia.org/rMW5472c66d7831f9e96adc663a9d1df077f23a1d5e) so it is not working at all. Perhaps there is a rather quick fix for it.

**Comment by s7eph4n** [comment]
This needs a rewrite of `ext.srf.slideshow.js` to use something more recent than `mw.legacy.ajax`.

**Comment by s7eph4n** [comment]
It also has to work under MW 1.19 and PHP 5.3 :nauseated_face: 

https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/master/INSTALL.md#platform-compatibility

**Comment by kghbln** [comment]
> This needs a rewrite of ext.srf.slideshow.js to use something more recent than mw.legacy.ajax.

Go for it if you have time. :)

> It also has to work under MW 1.19 and PHP 5.3

People using these versions currently do not have issues since it is working for them. Still I think it will be better to have a working solution for people who are actually upgrading their wikis as they should do. SRF 2.5.0 will drop compat with < MW 1.23 and < PHP 5.5 so a fix in master will be cool I guess.

**Comment by s7eph4n** [comment]
That was actually easier than I thought. I am not even sure, that the legacy module was ever needed.

@jaideraf Would be great if you could give this a spin.

**Comment by kghbln** [comment]
@s7eph4n Utterly cool! Thanks a bunch. Just to make sure. Will this be breaking for versions below MW 1.23 or does this even work for older versions?

**Comment by s7eph4n** [comment]
I did not test it, but I did some digging now and it turns out, that the dependency on `mw.legacy.ajax` was really just a leftover from an earlier (2012!) fix: https://gerrit.wikimedia.org/r/#/c/30442/

The added dependency on `mw.util` should not be a problem either. That module is already available in MW 1.19. In fact that it was missing was a bug that only did not break things because many other modules depend on it as well, so it was present in most cases anyway and nobody noticed.

Long story short: I am reasonably confident that it works with MW 1.19.

**Comment by kghbln** [comment]
Thanks a lot for these details!

> Long story short: I am reasonably confident that it works with MW 1.19.

So we could even do a SRF 2.4.2 earlier as soon as @jaideraf successfully tested. 

@mwjames These build failures seem to be unrelated and not a holdup?

**Comment by jaideraf** [comment]
@s7eph4n @kghbln I would test it, but I simply don't know what I have to do in `composer.json` to fetch this...
I tried `"dev-s7eph4n:slideshow"`, but it's not right. Any hint?
I am using a MW 1.28.0 now, I can't test it in 1.19.

**Comment by kghbln** [comment]
Indeed Composer is extremely painful when it comes to testing unmerged commits. To my experience it is an utter mess. However according to [this topic](http://stackoverflow.com/questions/25878984/can-i-pull-a-specific-commit-with-composer) it should be `mediawiki/semantic-result-formats "dev-master#0500fd4"` Keeping fingers crossed.

**Comment by jaideraf** [comment]
Not working in MW 1.28.0. :disappointed:

Test: http://wikincat.org/wiki/User:Jaider/Livros/Slideshow

Chrome says:

```
Uncaught Error: Unknown dependency: mediawiki.legacy.ajax
    at sortDependencies (load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:158)
    at sortDependencies (load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:159)
    at String.<anonymous> (load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:159)
    at Function.each (load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:5)
    at resolve (load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:159)
    at Object.load (load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:170)
    at Slideshow:194
    at startUp (load.php?debug=false&lang=en&modules=startup&only=scripts&skin=vector:30)
    at HTMLScriptElement.script.onload.script.onreadystatechange (load.php?debug=false&lang=en&modules=startup&only=scripts&skin=vector:30)
sortDependencies @ load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:158
sortDependencies @ load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:159
(anonymous) @ load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:159
each @ load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:5
resolve @ load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:159
load @ load.php?debug=false&lang=en&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0bjqe1c:170
(anonymous) @ Slideshow:194
startUp @ load.php?debug=false&lang=en&modules=startup&only=scripts&skin=vector:30
script.onload.script.onreadystatechange @ load.php?debug=false&lang=en&modules=startup&only=scripts&skin=vector:30
```

**Comment by kghbln** [comment]
I am pretty sure that you checked out master and not Stephans commit. This is what just happened to me. I have no idea whatsoever on how to fetch this commit. This is what I meant with utter mess.

**Comment by s7eph4n** [comment]
I don't think it is necessary to work with composer. Just go to the extensions directory and run

```
git checkout -b s7eph4n-slideshow master
git pull https://github.com/s7eph4n/SemanticResultFormats.git slideshow
```

**Comment by kghbln** [comment]
> I don't think it is necessary to work with composer. Just go to the extensions directory and run

Thanks a bunch for this tip. Worked great and [slideshow](https://www.semantic-mediawiki.org/wiki/Help:Slideshow_format) works great again, too!

**Comment by jaideraf** [comment]
Just to clarify: 

> _Just go to the extensions directory and run_

It means: Just go to the [SRF] **extension** directory and run

If you "_go to the_ [MW] _**extensions directory** and run_" those commands you mess all things there. :worried:

**Comment by s7eph4n** [comment]
Sorry!

**Comment by kghbln** [comment]
Let's get this out to the public: https://github.com/SemanticMediaWiki/SemanticResultFormats/pull/192 Thanks for your help!!!
